internal/static/integrations/windows_exporter/config_windows.go

```diff
@@ -1,55 +1,78 @@
+	errs = append(errs, err)
+	cfg.SMTP.ServerExclude, err = regexp.Compile(coalesceString(c.SMTP.Exclude, c.SMTP.BlackList))
+	errs = append(errs, err)
 
 	cfg.Textfile.TextFileDirectories = strings.Split(c.TextFile.TextFileDirectory, ",")
 
-	cfg.PhysicalDisk.DiskInclude = regexp.MustCompile(c.PhysicalDisk.Include)
-	cfg.PhysicalDisk.DiskExclude = regexp.MustCompile(c.PhysicalDisk.Exclude)
+	cfg.PhysicalDisk.DiskInclude, err = regexp.Compile(c.PhysicalDisk.Include)
+	errs = append(errs, err)
+	cfg.PhysicalDisk.DiskExclude, err = regexp.Compile(c.PhysicalDisk.Exclude)
+	errs = append(errs, err)
 
-	cfg.Printer.PrinterInclude = regexp.MustCompile(c.Printer.Include)
-	cfg.Printer.PrinterExclude = regexp.MustCompile(c.Printer.Exclude)
+	cfg.Printer.PrinterInclude, err = regexp.Compile(c.Printer.Include)
+	errs = append(errs, err)
+	cfg.Printer.PrinterExclude, err = regexp.Compile(c.Printer.Exclude)
+	errs = append(errs, err)
 
-	cfg.Process.ProcessExclude = regexp.MustCompile(coalesceString(c.Process.Exclude, c.Process.BlackList))
-	cfg.Process.ProcessInclude = regexp.MustCompile(coalesceString(c.Process.Include, c.Process.WhiteList))
+	cfg.Process.ProcessExclude, err = regexp.Compile(coalesceString(c.Process.Exclude, c.Process.BlackList))
+	errs = append(errs, err)
+	cfg.Process.ProcessInclude, err = regexp.Compile(coalesceString(c.Process.Include, c.Process.WhiteList))
+	errs = append(errs, err)
 
-	cfg.Net.NicExclude = regexp.MustCompile(coalesceString(c.Network.Exclude, c.Network.BlackList))
-	cfg.Net.NicInclude = regexp.MustCompile(coalesceString(c.Network.Include, c.Network.WhiteList))
+	cfg.Net.NicExclude, err = regexp.Compile(coalesceString(c.Network.Exclude, c.Network.BlackList))
+	errs = append(errs, err)
+	cfg.Net.NicInclude, err = regexp.Compile(coalesceString(c.Network.Include, c.Network.WhiteList))
+	errs = append(errs, err)
 
 	cfg.Mssql.CollectorsEnabled = strings.Split(c.MSSQL.EnabledClasses, ",")
 
 	cfg.Msmq.QueryWhereClause = &c.MSMQ.Where
 
-	cfg.LogicalDisk.VolumeInclude = regexp.MustCompile(coalesceString(c.LogicalDisk.Include, c.LogicalDisk.WhiteList))
-	cfg.LogicalDisk.VolumeExclude = regexp.MustCompile(coalesceString(c.LogicalDisk.Exclude, c.LogicalDisk.BlackList))
+	cfg.LogicalDisk.VolumeInclude, err = regexp.Compile(coalesceString(c.LogicalDisk.Include, c.LogicalDisk.WhiteList))
+	errs = append(errs, err)
+	cfg.LogicalDisk.VolumeExclude, err = regexp.Compile(coalesceString(c.LogicalDisk.Exclude, c.LogicalDisk.BlackList))
+	errs = append(errs, err)
 
-	cfg.ScheduledTask.TaskInclude = regexp.MustCompile(c.ScheduledTask.Include)
-	cfg.ScheduledTask.TaskExclude = regexp.MustCompile(c.ScheduledTask.Exclude)
+	cfg.ScheduledTask.TaskInclude, err = regexp.Compile(c.ScheduledTask.Include)
+	errs = append(errs, err)
+	cfg.ScheduledTask.TaskExclude, err = regexp.Compile(c.ScheduledTask.Exclude)
+	errs = append(errs, err)
 
-	return cfg
+	return cfg, errors.Join(errs...)
```

Comment:
That's cool - I didn't know there is a `Join` function :)
Personally I wouldn't return an object if there are errors (I'd probably return a nil instead), but I think it's ok if we do.

Reply:
If all values in the slice are `nil`, then nil is returned. Otherwise, an error is returned.

https://github.com/golang/go/blob/7c54e024e892cbb8ed24274f0b22505356220f10/src/errors/join.go#L20-L28

Reply:
The reason to choice join here was that if there are multiple errors, all errors returned instead the first one